packages/react-data-grid-addons/src/cells/headerCells/filters/NumericFilter.tsx

```diff
@@ -0,0 +1,148 @@
+import React from 'react';
+import { Column, RowData } from 'react-data-grid';
+
+enum RuleType {
+  Number = 1,
+  Range = 2,
+  GreaterThen = 3,
+  LessThen = 4
+}
+
+type Rule =
+  | { type: RuleType.Range; begin: number; end: number }
+  | { type: RuleType.GreaterThen | RuleType.LessThen | RuleType.Number; value: number };
+
+interface ChangeEvent {
+  filterTerm: Rule[] | null;
+  column: Column;
+  rawValue: string;
+  filterValues: typeof filterValues;
+}
+
+interface Props {
+  column: Column;
+  onChange(event: ChangeEvent): void;
+}
+
+export default function NumericFilter({ column, onChange }: Props) {
+  /** Validates the input */
+  function handleKeyPress(event: React.KeyboardEvent<HTMLInputElement>) {
+    const result = /[><,0-9-]/.test(event.key);
+    if (result === false) {
+      event.preventDefault();
+    }
+  }
+
+  function handleChange(event: React.ChangeEvent<HTMLInputElement>) {
+    const value = event.target.value;
+    const filters = getRules(value);
+    onChange({
+      filterTerm: filters.length > 0 ? filters : null,
+      column,
+      rawValue: value,
+      filterValues
+    });
+  }
+
+  const inputKey = `header-filter-${column.key}`;
+  const columnStyle: React.CSSProperties = {
+    float: 'left',
+    marginRight: 5,
+    maxWidth: '80%'
+  };
+  const badgeStyle: React.CSSProperties = {
+    cursor: 'help'
+  };
+
+  const tooltipText = 'Input Methods: Range (x-y), Greater Then (>x), Less Then (<y)';
+
+  return (
+    <div>
+      <div style={columnStyle}>
+        <input
+          key={inputKey}
```

Comment:
missing `type="text"`

Reply:
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input

>If this attribute is not specified, the default type adopted is `text`.

I can add it, but it's not necessary. wdyt?

Reply:
I usually prefer to add it to be explicit but it is fine in this case